js/packages/openinference-core/src/trace/trace_config/OpenInferenceTracer.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  context,
+  Context,
+  Span,
+  SpanOptions,
+  Tracer,
+} from "@opentelemetry/api";
+import {
+  OpenInferenceActiveSpanCallback,
+  TraceConfig,
+  TraceConfigOptions,
+} from "./types";
+import { OpenInferenceSpan } from "./OpenInferenceSpan";
+import { generateTraceConfig } from "./traceConfig";
+
+/**
+ * Formats the params for the startActiveSpan method
+ * The method has multiple overloads, so we need to format the arguments
+ * Taken from @see https://github.com/open-telemetry/opentelemetry-js/blob/main/packages/opentelemetry-sdk-trace-base/src/Tracer.ts#L220C3-L235C6
+ *
+ */
+function formatStartActiveSpanParams<F extends OpenInferenceActiveSpanCallback>(
+  arg2?: SpanOptions | F,
+  arg3?: Context | F,
+  arg4?: F,
+) {
+  let opts: SpanOptions | undefined;
+  let ctx: Context | undefined;
+  let fn: F;
+
+  if (arguments.length < 2) {
+    return;
+  } else if (arguments.length === 2) {
+    fn = arg2 as F;
+  } else if (arguments.length === 3) {
+    opts = arg2 as SpanOptions | undefined;
+    fn = arg3 as F;
+  } else {
+    opts = arg2 as SpanOptions | undefined;
+    ctx = arg3 as Context | undefined;
+    fn = arg4 as F;
+  }
+
+  opts = opts ?? {};
+  ctx = ctx ?? context.active();
+
+  return { opts, ctx, fn };
+}
+
+export class OpenInferenceTracer implements Tracer {
```

Comment:
I think in python this is OITracer to not confuse in other packages. Is there also no base class?

Reply:
I can update the name, pretty much same comment here as above in terms of base class, Didn't want to tie to any one implementation and instead just compose with whatever the tracer is that is provided in each instrumentation, felt the least intrusive